DataScience/Experimentation.py

```diff
@@ -406,24 +420,32 @@ def main(args):
             best_command = results[0]
             best_commands.append(['Hyper2', results[0]])
 
-        # TODO: Repeat above process of tuning parameters and interactions until convergence / no more improvements.
-
     t1 = datetime.now()
     print("\n\n*************************")
     Logger.info("Best parameters found after {}:".format((t1-t0)-timedelta(microseconds=(t1-t0).microseconds)))
     best_command.prints()
     print("*************************")
 
+    # TODO: Repeat above process of tuning parameters and interactions until convergence / no more improvements.
+
+    t2 = datetime.now()
```

Comment:
generate predictions files time  will get added to the generate model time in this case.

Reply:
Generate model time is t3-t2 (from line 442). If args.generate_predictions is present, then we reset t2 on line 434. So in either case, I see generate model time is for generate model time alone.